layers.go

```diff
@@ -1156,23 +1153,24 @@ func (s *store) newLayerStore(rundir, layerdir, imagedir string, driver drivers.
 }
 
 func newROLayerStore(rundir string, layerdir string, driver drivers.Driver) (roLayerStore, error) {
-	lockfile, err := lockfile.GetROLockFile(filepath.Join(layerdir, "layers.lock"))
+	roLockfile, err := lockfile.GetROLockFile(filepath.Join(layerdir, "layers.lock"))
 	if err != nil {
 		return nil, err
 	}
+	// Ignore errors if the lock file could not be opened
+	mountsLockfile, _ := lockfile.GetLockFile(filepath.Join(rundir, "mountpoints.lock"))
```

Comment:
Reading `getROLayerStoresLocked`, `rundir` is the same for _all_ RO layer stores, so this doesn’t work; they are going to stomp over each others’ data.

And if we assume that two concurrent processes could share the same layer store on the same system, one process as primary, the second as a read-only additional … this just doesn’t work. _And_ I can’t see a upgrade/migration path for processes currently using the `rundir`-based locations.